app/Http/Controllers/DashboardController.php

```diff
@@ -20,30 +23,41 @@ public function __construct()
     public function store(Request $req)
     {
         $this->validate($req, [
-            'expAmount' => 'required|numeric',
+            'expAmount' => 'required|numeric|min:1',
             'expType' => 'required|max:255',
-            'expDate' => 'required|date',
-            'expComments' => 'nullable|max:255'
+            'expDate' => 'required|date'
         ]);
         $newExpense = new Expense;
         $newExpense->owner_username= Auth::user()->username;
         $newExpense->amount=$req->expAmount;
         $newExpense->type=$req->expType;
         $newExpense->date_added=$req->expDate;
         $newExpense->comments=$req->expComments;
+
+        // TODO Refactor to not save an expense row when shared expenses has errors
         if($newExpense->save()) {
-            if(isset($req->expOwedAmount)&&isset($req->expOwerUsername)) {
-                $this->validate($req, [
-                    'expOwedAmount' => 'required|numeric',
-                    'expOwerUsername' => 'required|max:255',
-                    'expOwerComments' => 'nullable|max:255'
-                ]);
-                $newSharedExpense = new SharedExpense;
-                $newSharedExpense->expense_id=$newExpense->id;
-                $newSharedExpense->amount_owed=$req->expOwedAmount;
-                $newSharedExpense->comments=$req->expOwerComments;
-                $newSharedExpense->secondary_username=$req->expOwerUsername;
-                $newSharedExpense->save();
+            if(isset($req->username) || isset($req->expOwedAmount)) {
+
+              // TODO Set comment to nullable in database
+              $this->validate($req, [
+                  'expOwedAmount'   => 'required|numeric|min:1',
+                  'username'        => 'exists:users|required|max:255',
```

Comment:
i put username to match the column in the users table. other column names throws an error